modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -118,9 +121,20 @@ private ClientListenerResponse handle0(ClientListenerRequest req) {
         if (req0.isAsync(ctx)) {
             IgniteInternalFuture<ClientResponse> fut = req0.processAsync(ctx);
 
+            if (!fut.isDone()) {
+                try {
+                    // Give request a chance to be executed and response processed by the current thread,
+                    // so we can avoid any performance drops caused by async requests execution.
+                    fut.get(ASYNC_REQUEST_WAIT_TIMEOUT);
```

Comment:
Is it a good idea to block the IO thread, even for a short period? 

Do we have any benchmark results for this change?

Reply:
@ptupitsyn, it's not IO thread, it's client-connector thread. Before IGNITE-21183 these threads are always blocked, and it seems to be ok except the cases with waiting on locked keys. Now we wait extra time only on locked keys, in other cases we release the thread as soon as result become available. 
Yes, I have benchmark results for this change, performance drop is fixed now. I will publish results today as a comment to the ticket.

Reply:
> not IO thread, it's client-connector thread

`client-connector` does IO, that's what I had in mind.